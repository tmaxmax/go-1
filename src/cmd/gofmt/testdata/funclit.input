//gofmt -s

// Copyright 2022 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.
// Test for function literal lightening.

package p

func _() {
	f(func() {})
	f(func(x int) {})
	f(func(x, y, z string) {})
	f(func(x, y, z string, a, b int) {})
	f(func() int {})
	f(func() (int, string) {})
	f(func() (_, _ int, _ string) {})
	f(func(x int) int {})
	f(func(x, y, z string) (int, string) {})
	f(func(x, y, z string, a, b int) (_, _ int, _ string) {})
}

func _() {
	f = func() {}
	f = func(x int) {}
	f = func(x, y int) bool { return x < y }
	f = func(x int) {fmt.Println("Hello there, this should surely go way over the allowed single-line character limit...")}
	f = func(x, y int) bool {
		x = y ^ x
		y = x ^ y
		x = y ^ x
		return x < y
	}
	f, g, h = 1, func(x, y int){}, 2
	_ = func(x int) {} // cannot be simplified
	f, _, g = 1, func(x, y int){}, 2
}