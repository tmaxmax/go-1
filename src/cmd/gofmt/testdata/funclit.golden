//gofmt -s

// Copyright 2022 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.
// Test for function literal lightening.

package p

func _() {
	f(func {})
	f(func { x | })
	f(func { x, y, z | })
	f(func { x, y, z, a, b | })
	f(func {})
	f(func {})
	f(func {})
	f(func { x | })
	f(func { x, y, z | })
	f(func { x, y, z, a, b | })
}

func _() {
	f = func() {}
	f = func { x | }
	f = func { x, y | return x < y }
	f = func { x |
		fmt.Println("Hello there, this should surely go way over the allowed single-line character limit...")
	}
	f = func { x, y |
		x = y ^ x
		y = x ^ y
		x = y ^ x
		return x < y
	}
	f, g, h = 1, func { x, y | }, 2
	_ = func(x int) {} // cannot be simplified
	f, _, g = 1, func(x, y int) {}, 2
}
